### ISSUE TITLE
IDE0036 not raised for `protected private` and `internal protected` members

### ISSUE THREAD

**Issue by Arthri** [dotnet/roslyn#73768]
**Version Used**: 4.9.0-3.24121.1 (a98c90d5) | C#12

**Steps to Reproduce**:
1. Declare an `internal protected` member
2. Declare a `protected private` member
3. Use the following .editorconfig settings
  ```editorconfig
  [*.cs]
  dotnet_diagnostic.IDE0036.severity = warning
  csharp_preferred_modifier_order = public, private, protected, internal, file, static, extern, new, virtual, abstract, sealed, override, readonly, unsafe, required, volatile, async
  ```

[sharplab](https://sharplab.io/#v2:CYLg1APgdghgtgUwM4AcYGMEAIBaBuAWAChiBLKAFwQCdYAbLAAQCYsANYgb2K163Kq0YDFNQD2VdFWBNWATR59uRPlgC+xRb1ESEUhDNGkAbjCpMALFgAyACgCUWrMtUaiboA==)

**Diagnostic Id**: `"IDE0036: Order modifiers"`

**Expected Behavior**: IDE0036 should be raised because `protected private` and `internal protected` violate the order

**Actual Behavior**: IDE0036 not raised

**Comment by Youssef1313** [comment]
It looks like the severity set for `IDE0036` isn't taken into account. I can get the warning with:

```ini
csharp_preferred_modifier_order = public, private, protected, internal, file, static, extern, new, virtual, abstract, sealed, override, readonly, unsafe, required, volatile, async:warning
```

**Comment by akhera99** [comment]
This no longer seems to reproduce, closing out.
